From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joey Arhar <jarhar@chromium.org>
Date: Thu, 8 Oct 2026 17:33:30 -0700
Subject: [M155] Avoid dangling reference in Element::CloneAttributesFrom()

Original change's description:
> Avoid dangling reference in Element::CloneAttributesFrom()
>
> Copy Attribute by value in Element::CloneAttributesFrom() instead of
> holding a reference into the element's attribute storage, which may be
> modified during AttributeChangedWithInvalidations().
>
> There will be a test for this in a subsequent patch. It isn't included
> here because it hits a DCHECK.
>
> Fixed: 568730794
> Change-Id: I4050cecff5753ebe435fd87ac1686e416f065c35
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8508875
> Reviewed-by: Joey Arhar <jarhar@chromium.org>
> Commit-Queue: Joey Arhar <jarhar@chromium.org>
> Reviewed-by: David Baron <dbaron@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1711816}

(cherry picked from commit bdc63e2cce0b3abc7dc984eb7f4e5168990554b2)

Bug: 571383444,568730794
Change-Id: I4050cecff5753ebe435fd87ac1686e416f065c35
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8543393
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/8059@{#1056}
Cr-Branched-From: 199ac2cd7ab289e11c2236bd2cede848781213da-refs/heads/main@{#1697595}

diff --git a/third_party/blink/renderer/core/dom/element.cc b/third_party/blink/renderer/core/dom/element.cc
index c81c41d3b1c653bfd016e2e52aa3186dd70b949e..70cd411dac83cde528f01f6c51d551ad3446ec20 100644
--- a/third_party/blink/renderer/core/dom/element.cc
+++ b/third_party/blink/renderer/core/dom/element.cc
@@ -11910,7 +11910,7 @@ void Element::CloneAttributesFrom(const Element& other) {
   }
 
   for (wtf_size_t i = 0; i < element_data_->Attributes().size(); ++i) {
-    const Attribute& attr = element_data_->Attributes().at(i);
+    const Attribute attr = element_data_->Attributes().at(i);
     attribute_or_class_bloom_ |= FilterForAttribute(attr.GetName());
     AttributeChanged(
         AttributeModificationParams(attr.GetName(), g_null_atom, attr.Value(),
